CMSgov/design-system
# [WNMGDS-1679] Bugfix: tooltip not going away when focus lost

## Summary

- Currently focusing on a tooltip trigger will open the tooltip but shifting focus will not close it automatically.
- Changes logic to check if `event.currentTarget !== event.target` to close when focus is lost.
- Adds unit tests for this functionality

## How to test

1. `yarn install && yarn build && yarn storybook`
2. Open the tooltip examples and use keyboard to navigate them, moving focus away from a trigger should close the tooltip.
3. `yarn test:unit`

## Checklist
